### PR TITLE
Added more ZDC functionality to DQM

### PR DESCRIPTION
**Pull request by cfmcginn**
#### PR description:

This PR is to replace existing, closed PR here: https://github.com/cms-sw/cmssw/pull/45948

The PR adds to the DQM for ZDC specific histograms, via ZDCQIE10Task

This PR is updated relative to the closed previous PR, mostly fixing build issue, retitling some histograms, and updating python such that emulated ZDC sums fill. Another PR exists where it is rebased to 14_1_X for backport, currently open  here https://github.com/cms-sw/cmssw/pull/46407.

#### PR validation:

The PR is tested in CMSSW_14_2_X_2024-10-20-0000, after running a merge-topic to incorporate the PR into a fresh area
scram b ran successfully, validating the build

Following test command produces the expected output:
```
cmsRun DQM/Integration/python/clients/hcal_dqm_sourceclient-live_cfg.py inputFiles=/eos/cms/store/group/dpg_hcal/comm_hcal/PFG/backup_raw/HIHLTPhysics-HIRun2023A-RAW-v1-375754.root runkey=hi_run
```
under the upload directory, if using the offline candidate GT. Inspected histograms inside the DQM file appear reasonable for ZDC sums.

In addition, the following set of commands are run
scram b runtests
scram build code-checks
scram build code-format

code-checks revealed no issues, code-format offered no suggestions. runtests had multiple failures, detailed below - however, every failure existed already with runtests in a fresh area, and matches the set of failures documented already in the backport. As such, it is unclear the failures are the result of the PR

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

As mentioned above, this PR is to replace the closed PR https://github.com/cms-sw/cmssw/pull/45948, and is paired to the backport to CMSSW_14_1_X, found here https://github.com/cms-sw/cmssw/pull/46407

### CODE REVIEW

**Comment by cfmcginn** [comment]
@hjbossi @mandrenguyen

**Comment by antoniovagnerini** [comment]
please test

File: DQM/Integration/python/clients/hcal_dqm_sourceclient-live_cfg.py [hunk 1]
```diff
@@ -32,10 +32,11 @@
 #-------------------------------------
 from DQM.Integration.config.online_customizations_cfi import *
 if useOfflineGT:
-	process.load("Configuration.StandardSequences.FrontierConditions_GlobalTag_cff")
-	process.GlobalTag.globaltag = autoCond['run3_data_prompt'] 
+        process.load("Configuration.StandardSequences.FrontierConditions_GlobalTag_cff")
+        process.GlobalTag.globaltag = '141X_dataRun3_Prompt_Candidate_2024_10_08_09_42_50'
```
Comment (mmusich) [inline DQM/Integration/python/clients/hcal_dqm_sourceclient-live_cfg.py:36]: none of the comments of the 14.1.X backport was addressed.

Comment (cfmcginn) [reply DQM/Integration/python/clients/hcal_dqm_sourceclient-live_cfg.py:36]: I opted to get this up such that it was faithful to the backport version code - happy to address the comments now, will go thru and recommit

Comment (cfmcginn) [reply DQM/Integration/python/clients/hcal_dqm_sourceclient-live_cfg.py:36]: Adding: I will go thru and update this PR and the backport simultaneously

Comment (cfmcginn) [reply DQM/Integration/python/clients/hcal_dqm_sourceclient-live_cfg.py]: Responded issue-by-issue in backport thread, all suggestions adopted, both PRs updated

**Comment by antoniovagnerini** [comment]
please test

File: DQM/Integration/python/clients/hcal_dqm_sourceclient-live_cfg.py [hunk 2]
```diff
@@ -94,27 +97,34 @@
...
+
 process.hcalDigis.InputLabel = rawTag
 process.emulTPDigisNoTDCCut = process.emulTPDigis.clone(
      parameters = cms.untracked.PSet(
 	ADCThresholdHF = cms.uint32(255),
 	TDCMaskHF = cms.uint64(0xFFFFFFFFFFFFFFFF)
      )
 )
-process.HcalTPGCoderULUT.LUTGenerationMode = False
+process.HcalTPGCoderULUT.LUTGenerationMode = True
```
Comment (mmusich) [inline DQM/Integration/python/clients/hcal_dqm_sourceclient-live_cfg.py:127]: is this ultimately really needed given the answer at https://github.com/cms-sw/cmssw/pull/46407/files#r1804677477 ?

Comment (hjbossi) [reply DQM/Integration/python/clients/hcal_dqm_sourceclient-live_cfg.py:127]: This should be reverted to false, ZDC channels were now added to HcalL1TriggerObjects, fixing the need to set this to true.

Comment (cfmcginn) [reply DQM/Integration/python/clients/hcal_dqm_sourceclient-live_cfg.py:127]: I can make the reversion to False requested, but I would like to also understand the failed check first and fix everything in one go - if I click through on the error details I can't actually navigate the page after to find anything. Any guidance on where to look to fix this failed check is appreciated

Comment (mmusich) [reply DQM/Integration/python/clients/hcal_dqm_sourceclient-live_cfg.py:127]: > I would like to also understand the failed check first and fix everything in one go

nothing failed, these were transient das errors.

Comment (cfmcginn) [reply DQM/Integration/python/clients/hcal_dqm_sourceclient-live_cfg.py:127]: Ah! Apologies, I did not understand that - I will revert to False

**Comment by mmusich** [comment]
please test

File: DQM/HcalTasks/interface/ZDCQIE10Task.h [hunk 3]
```diff
@@ -9,24 +9,83 @@
  */
 
 #include "DQM/HcalCommon/interface/ElectronicsMap.h"
+#include "DQM/HcalTasks/interface/DigiTask.h"
```
[the remark this answers is not in the document]
Comment (cfmcginn) [reply DQM/HcalTasks/interface/ZDCQIE10Task.h]: It appears this is also an unnecessary header, and is removed

File: DQM/HcalTasks/interface/ZDCQIE10Task.h [hunk 4]
```diff
@@ -9,24 +9,83 @@
...
+#include "DQM/HcalCommon/interface/Container2D.h"
+#include "DQM/HcalCommon/interface/ContainerProf1D.h"
+#include "DQM/HcalCommon/interface/ContainerProf2D.h"
+#include "DQM/HcalCommon/interface/ContainerSingle1D.h"
+#include "DQM/HcalCommon/interface/ContainerSingle2D.h"
+#include "DQM/HcalCommon/interface/ContainerSingleProf2D.h"
+#include "DQM/HcalCommon/interface/Constants.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
```
[the remark this answers is not in the document]
Comment (cfmcginn) [reply DQM/HcalTasks/interface/ZDCQIE10Task.h]: Appears not needed - removed in latest commit

File: DQM/HcalTasks/interface/ZDCQIE10Task.h [hunk 5]
```diff
@@ -9,24 +9,83 @@
...
+#include "DataFormats/HcalDigi/interface/HcalDigiCollections.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
 
-class ZDCQIE10Task : public DQMEDAnalyzer {
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
```
[the remark this answers is not in the document]
Comment (cfmcginn) [reply DQM/HcalTasks/interface/ZDCQIE10Task.h]: Unneeded - removed in latest commit

File: DQM/Integration/python/clients/hcal_dqm_sourceclient-live_cfg.py (hunk 2 above)
Comment (cfmcginn) [reply DQM/Integration/python/clients/hcal_dqm_sourceclient-live_cfg.py]: Reverted to false in latest commit

**Comment by cfmcginn** [comment]
If useful, just want to note that to run the offline test command
```
 cmsRun DQM/Integration/python/clients/hcal_dqm_sourceclient-live_cfg.py inputFiles=/eos/cms/store/group/dpg_hcal/comm_hcal/PFG/backup_raw/HIHLTPhysics-HIRun2023A-RAW-v1-375754.root runkey=hi_run
```
I have to make some changes to the python (things we have reverted like specifying the candidate global tag), but with the latest commits removing all the headers it still runs and produces what appears to be reasonable output

**Comment by antoniovagnerini** [comment]
+1
- spurious differences from HGCal DQM test (open issue  #46416)

**Comment by mandrenguyen** [comment]
+1

**Pull request merged**